NetworkConfigurations/RoutingAndForwarding/OSPFv2.md

```diff
@@ -0,0 +1,298 @@
+
+### Installation using APT:
+
+```
+curl -s https://deb.frrouting.org/frr/keys.gpg | sudo tee /usr/share/keyrings/frrouting.gpg > /dev/null
+
+FRRVER="frr-stable"
+echo deb '[signed-by=/usr/share/keyrings/frrouting.gpg]' https://deb.frrouting.org/frr \
+     $(lsb_release -s -c) $FRRVER | sudo tee -a /etc/apt/sources.list.d/frr.list
+
+sudo apt update && sudo apt install frr frr-pythontools
+```
+
+Note: Ensure that `apt-transport-https` and `curl` are installed before proceeding with the installation.
+
+### Installation using Snapcraft (Alternative Method):
+
+```
+sudo apt update
+sudo apt install snapd
+sudo snap install core
+
+sudo snap install frr
+```
+
+### Start and Enable FRR Service:
+
+```
+sudo systemctl start frr
+sudo systemctl enable frr
+```
+
+Output-
+
+```
+Synchronizing state of frr.service with SysV service script with /lib/systemd/systemd-sysv-install.
+Executing: /lib/systemd/systemd-sysv-install enable frr
+```
+
+## Example Configuration
+
+### Enable OSPFd:
+
+Edit the FRR daemon configuration file on both the devices:
+
+```
+sudo nano /etc/frr/daemons
+```
+
+Ensure the following configuration:
+
+```
+zebra=yes
+bgpd=no
+ospfd=yes
+ospf6d=no
+ripd=no
+ripngd=no
+```
+
+<p align="center">
+  <img src="../../Images/ImagesForNetworkConfiguration/OSPFv2.png" alt="OSPFv2">
+</p>
+
+### Assign IP Addresses:
+
+On Device 1 (dent3.2-1):
+
+```
+ip addr add 192.168.1.3/24 dev enp0s4
+ip link set dev enp0s4 up
+```
+
+On Device 2 (dent3.2-2):
+
+```
+ip addr add 192.168.1.2/24 dev enp0s4
+ip link set dev enp0s4 up
+```
+
+Ensure IPv4 forwarding is enabled on each device (`sysctl -w net.ipv4.ip_forward=1`).
+
+### Configure OSPF:
+
+Edit the FRR configuration file on both the devices:
+
+```
+sudo nano /etc/frr/frr.conf
+```
+
+Add the following configuration to the file:
+
+```
+! Enable OSPF routing process
+router ospf
+
+! Configure OSPF on the enp0s4 interface
+interface enp0s4
+  ip ospf area 0
+```
+
+**Note:** The configuration may vary depending on user requirements.
+
+### Restart FRR:
+
+```
+sudo systemctl restart frr
+```
+
+### Interacting with OSPF:
+
+To access the interactive OSPF shell:
+
+```
+sudo vtysh
+```
+
+Output-
+
+```
+Hello, this is FRRouting (version 8.5).
+Copyright 1996-2005 Kunihiro Ishiguro, et al.
+```
+
+After accessing the `vtysh` shell, you can execute OSPF-specific commands such as `show ip ospf`, `show ip ospf neighbor`, and `show ip route` to inspect OSPF configurations and neighbor relationships.
+
+- `show ip ospf`: Displays the current status and configuration details of the OSPF routing process.
+- `show ip ospf neighbor`: Lists the OSPF neighbor relationships and their statuses.
+- `show ip route`: Shows the current routing table and all the IP routes known to the device.
+
+For more information on these and other available commands, please refer to [OSPFv2 FRR](https://docs.frrouting.org/en/latest/ospfd.html).
+
```

Comment:
I do not know if the topology is a good one to demonstrate OSPF. There is only one way for dent3.2-1 to connect to Cloud2 (similarly there is only one way for dent3.2-1 to connect to Cloud1). I think we should add another path for the switches to connect to "other" cloud and show 1) OSPF really forwards the package through the shortest path, and 2) if the shortest path is not available, traffic goes through the longer path.